### DISCUSSION TITLE
Integrating with react native first and using design systems to cross plaform for web/ios/android/AR and etc.

### DISCUSSION THREAD

**Discussion by ebredy** [Budibase/budibase#3873] [Ideas]
Allow the ability to have new components added from different design systems:

1) Check this out: https://medium.com/@MisterKevin_js/lets-create-a-cross-platform-design-system-part-1-85019ebc9b8e
here is another medium article, we can create our own design system or/and use existing ones as options: https://medium.com/akveo-engineering/what-is-the-most-flexible-design-system-for-react-native-a1e7481dbf2a

2) Ideally integrating with existing ones can attract more developers who have an existing codebase. an approach might be to create some kind of feature that makes it easy for any dev to integrate a new design system as part of this lo/code platform: https://www.codeinwp.com/blog/react-native-component-libraries/ or manually accept pull requests for other devs to add new components from different design systems.

If it is something that can't be done then I'll have to pay resources to do this privately.  I am a full-stack developer myself who is very interested in the open source no code/lo-code ecosystem.

**Comment by mjashanks** [comment]
This is not possible today. We hope to add the ability to create custom components this year (hopefully in the first half). I'm not sure about using existing component libraries though, as Budibase already expects a certain API for components.

Thanks for this though - great info!
